.github/workflows/k6-action-image-release.yml

```diff
@@ -1,69 +1,31 @@
-name: Release K6 Action Image
+name: Scan/Release K6 Action Image
 
 on:
   push:
+    branches:
```

Comment:
Lines 5-10 are not supposed to be here right? We can't push to main in general.

Reply:
When we squash/merge a PR there will be a push to main. 
The reusable workflow then runs a build and scan, and if the input `release_latest` is set to true it will also push an image with the latest tag.
So for this workflow it just runs another build and scan whenever there are new changes on main

Reply:
And then you are adding the if statement in the reusable workflow: https://github.com/Altinn/altinn-platform/pull/1377/files#diff-0b86917293b396a7379a4c1cbb51a823c6bd8ce5a98db360fccd925ec0877674R107

Reply:
Brain knot..
When we create a release via the UI it will trigger the workflow based on
```
    tags:
      - "k6-action-image-*"
```
To trigger a release based on main we will need to flip the **_release_latest_** to **_true_**.

The scan gets triggered every time you change one of files captured within the paths block.

Reply:
Well, no, the branches and paths need to match both: https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#running-your-workflow-only-when-a-push-affects-specific-files

Reply:
The pull_request block will trigger scanning tho.

Reply:
> When we squash/merge a PR there will be a push to main. The reusable workflow then runs a build and scan, and if the input `release_latest` is set to true it will also push an image with the latest tag. So for this workflow it just runs another build and scan whenever there are new changes on main

Sorry, didn't notice you had replied already.
I think this might end up being problematic if new vulnerabilities are found. Won't we be prevented from merging to main as the build from main will always fail and we won't have any way to update the trivy ignore file in main?

Reply:
nvm, I guess we will have to fix it when we open a PR before merging to main

Reply:
Just a kind of "fail safe" when the PR code is merged with latest from main, just to verify that the latest commit on main with all changes are ok. We can't fix it on main and would then have to issue another PR to the main to fix the build.